nanachan/discord/bot.py

```diff
@@ -258,10 +258,21 @@ async def on_command_error(self, ctx, error, force=False):
                 await ctx.send_help(ctx.command)
         else:
             error_msg = await self.extract_error(error)
-            if not isinstance(error, IHateThe3SecondsTimeout):
+            # Check if this is an IHateThe3SecondsTimeout error (either directly or wrapped)
+            original_error = (
+                error.original if isinstance(error, commands.CommandInvokeError) else error
+            )
+            if not isinstance(original_error, IHateThe3SecondsTimeout):
                 await self.send_error(error_msg, source=ctx.message, reply=ctx)
 
     async def on_app_command_error(self, interaction: Interaction, error: AppCommandError):
+        # Check if this is an IHateThe3SecondsTimeout error (either directly or wrapped)
+        original_error = error.original if isinstance(error, CommandInvokeError) else error
+        if isinstance(original_error, IHateThe3SecondsTimeout):
+            # Log the error but don't try to respond since the interaction has expired
```

Comment:
sounds like it would make sense but actually doesn’t, afaik it doesn’t happen, probably `if interaction.response.is_done():` below catches that. if it were actually the case it wouldn’t send the error through the webhook.